pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -266,9 +263,9 @@ def _make_backward_fn(self, optimizer: Optimizer, opt_idx: int) -> Optional[Call
         def backward_fn(loss: Tensor):
             self.backward(loss, optimizer, opt_idx)
 
-            # check if loss or model weights are nan
             if self.trainer.terminate_on_nan:
-                check_finite_loss(self.trainer.lightning_module, loss)
+                # check if model weights are nan after backward
+                detect_nan_parameters(self.trainer.lightning_module)
```

Comment:
reviewer: checking the loss value only makes sense before backward. 
here we are only interested in checking the parameter values of the model.